namer/serversets/src/main/scala/io/buoyant/namer/serversets/ServersetNamer.scala

```diff
@@ -49,7 +54,12 @@ class ServersetNamer(zkHost: String, idPrefix: Path) extends Namer {
         val addr = if (segments.nonEmpty && (segments.last contains ":")) {
           val Array(name, endpoint) = segments.last.split(":", 2)
           val zkPath = (segments.init :+ name).mkString("/", "/", "")
-          resolveServerset(zkHost, zkPath, endpoint)
+          if (!(endpoint contains "#")) {
```

Comment:
This is a moot point if we use `ServersetPath` as I recommend above, but I'll just point it out anyway: Linkerd style is to avoid using infix notation.  So this should be `endpoint.contains("#")`.  Furthermore, I'd flip the `if` and `else` blocks to remove the negation in the condition.